src/callbacks/apply.js

```diff
@@ -7,6 +7,27 @@ import exec from '../utils/execHelper';
 const IS_WINDOWS = process.platform === 'win32';
 const ticks = IS_WINDOWS ? '"' : '\'';
 
+export const LFS_DEBUG = (() => {
+  let shouldRecordEvents = false;
+  let events = [];
+  return {
+    clearEvents: () => { events = []; },
+    dumpEvents: (dumpPath) => fse.writeJsonSync(dumpPath, events, { spaces: 2 }),
+    recordEvent: (event, eventData) => {
+      if (shouldRecordEvents) {
+        events.push({
+          event,
+          data: eventData
+        });
+      }
+    },
+    retrieveEvents: () => [...events],
```

Comment:
If this weren't a debug tool I would definitely have you adjust this, just fyi this is not equivalent to `fp.clone`. You are producing a new array reference, but each of the events here are referentially identical to the ones we have in the internal memory. This means that if you retrieveEvents and then modify any event in the list, you also modify the event in the internal memory.

This is definitely a big distinction, and you should look out for it on more critical code paths.